libcaf_core/caf/type_id.hpp

```diff
@@ -115,148 +198,118 @@ constexpr type_id_t first_custom_type_id = 200;
-  // CAF_ADD_ATOM(builtin, caf, pong_atom)
-  // CAF_ADD_ATOM(builtin, caf, publish_atom)
-  // CAF_ADD_ATOM(builtin, caf, publish_udp_atom)
-  // CAF_ADD_ATOM(builtin, caf, put_atom)
-  // CAF_ADD_ATOM(builtin, caf, receive_atom)
-  // CAF_ADD_ATOM(builtin, caf, redirect_atom)
-  // CAF_ADD_ATOM(builtin, caf, reset_atom)
-  // CAF_ADD_ATOM(builtin, caf, resolve_atom)
-  // CAF_ADD_ATOM(builtin, caf, spawn_atom)
-  // CAF_ADD_ATOM(builtin, caf, stream_atom)
-  // CAF_ADD_ATOM(builtin, caf, sub_atom)
-  // CAF_ADD_ATOM(builtin, caf, subscribe_atom)
-  // CAF_ADD_ATOM(builtin, caf, sys_atom)
-  // CAF_ADD_ATOM(builtin, caf, tick_atom)
-  // CAF_ADD_ATOM(builtin, caf, timeout_atom)
-  // CAF_ADD_ATOM(builtin, caf, unlink_atom)
-  // CAF_ADD_ATOM(builtin, caf, unpublish_atom)
-  // CAF_ADD_ATOM(builtin, caf, unpublish_udp_atom)
-  // CAF_ADD_ATOM(builtin, caf, unsubscribe_atom)
-  // CAF_ADD_ATOM(builtin, caf, update_atom)
-  // CAF_ADD_ATOM(builtin, caf, wait_for_atom)
+  CAF_ADD_ATOM(builtin, caf, add_atom)
+  CAF_ADD_ATOM(builtin, caf, close_atom)
+  CAF_ADD_ATOM(builtin, caf, connect_atom)
+  CAF_ADD_ATOM(builtin, caf, contact_atom)
+  CAF_ADD_ATOM(builtin, caf, delete_atom)
+  CAF_ADD_ATOM(builtin, caf, demonitor_atom)
+  CAF_ADD_ATOM(builtin, caf, div_atom)
+  CAF_ADD_ATOM(builtin, caf, flush_atom)
+  CAF_ADD_ATOM(builtin, caf, forward_atom)
+  CAF_ADD_ATOM(builtin, caf, get_atom)
+  CAF_ADD_ATOM(builtin, caf, idle_atom)
+  CAF_ADD_ATOM(builtin, caf, join_atom)
+  CAF_ADD_ATOM(builtin, caf, leave_atom)
+  CAF_ADD_ATOM(builtin, caf, link_atom)
+  CAF_ADD_ATOM(builtin, caf, migrate_atom)
+  CAF_ADD_ATOM(builtin, caf, monitor_atom)
+  CAF_ADD_ATOM(builtin, caf, mul_atom)
+  CAF_ADD_ATOM(builtin, caf, ok_atom)
+  CAF_ADD_ATOM(builtin, caf, open_atom)
+  CAF_ADD_ATOM(builtin, caf, pending_atom)
+  CAF_ADD_ATOM(builtin, caf, ping_atom)
+  CAF_ADD_ATOM(builtin, caf, pong_atom)
+  CAF_ADD_ATOM(builtin, caf, publish_atom)
+  CAF_ADD_ATOM(builtin, caf, publish_udp_atom)
+  CAF_ADD_ATOM(builtin, caf, put_atom)
+  CAF_ADD_ATOM(builtin, caf, receive_atom)
+  CAF_ADD_ATOM(builtin, caf, redirect_atom)
+  CAF_ADD_ATOM(builtin, caf, reset_atom)
+  CAF_ADD_ATOM(builtin, caf, resolve_atom)
+  CAF_ADD_ATOM(builtin, caf, spawn_atom)
+  CAF_ADD_ATOM(builtin, caf, stream_atom)
+  CAF_ADD_ATOM(builtin, caf, sub_atom)
+  CAF_ADD_ATOM(builtin, caf, subscribe_atom)
+  CAF_ADD_ATOM(builtin, caf, sys_atom)
+  CAF_ADD_ATOM(builtin, caf, tick_atom)
+  CAF_ADD_ATOM(builtin, caf, timeout_atom)
+  CAF_ADD_ATOM(builtin, caf, unlink_atom)
+  CAF_ADD_ATOM(builtin, caf, unpublish_atom)
+  CAF_ADD_ATOM(builtin, caf, unpublish_udp_atom)
+  CAF_ADD_ATOM(builtin, caf, unsubscribe_atom)
+  CAF_ADD_ATOM(builtin, caf, update_atom)
+  CAF_ADD_ATOM(builtin, caf, wait_for_atom)
 
 CAF_END_TYPE_ID_BLOCK(builtin)
+
+namespace caf::detail {
+
+static constexpr type_id_t io_module_begin = builtin_type_ids::end;
+
+static constexpr type_id_t io_module_end = io_module_begin + 19;
```

Comment:
Yes. There's a `static_assert` in `io/fwd.hpp` that checks whether the type ID block in I/O matches the begin/end IDs we declare here. So it can't break silently.